File: app/src/main/java/com/leadinsource/dailyweightlog/ui/main/PreviousFragment.java
package com.leadinsource.dailyweightlog.ui.main;


import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.leadinsource.dailyweightlog.R;

/**
 * A simple {@link Fragment} subclass.
 */
public class PreviousFragment extends Fragment {


    public PreviousFragment() {
        // Required empty public constructor
    }


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_previous, container, false);
    }

}

File: app/src/main/java/com/leadinsource/dailyweightlog/ui/main/PreviousFragment.kt
package com.leadinsource.dailyweightlog.ui.main


import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.databinding.ViewDataBinding

import com.leadinsource.dailyweightlog.R
import com.leadinsource.dailyweightlog.app.DWLApplication
import com.leadinsource.dailyweightlog.databinding.FragmentPreviousBinding
import com.leadinsource.dailyweightlog.ui.DataBindingFragment
import javax.inject.Inject


/**
 * A simple [Fragment] subclass.
 */
class PreviousFragment : DataBindingFragment() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        DWLApplication.app().appComponent().inject(this)
    }

    //viewModel shared with activity
    @Inject
    lateinit var viewModel: MainActivityViewModel

    override fun inflateBinding(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ) = FragmentPreviousBinding.inflate(inflater, container, false).also {
        it.viewModel = viewModel
    }

}// Required empty public constructor
